src/dd/dd.rs

```diff
@@ -0,0 +1,745 @@
+        write!(
+            f,
+            "if: {if_}\n\
+             of: {of_}\n\
+             ibs: {ibs}\n\
+             obs: {obs}\n\
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            if_ = self.inp_name,
+            of_ = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "ibs" => if let Some(m) = get_u(val) {
+                    cf.inp_block_size = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "obs" => if let Some(m) = get_u(val) {
+                    cf.out_block_size = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "bs" => if let Some(m) = get_u(val) {
+                    cf.inp_block_size = m;
+                    cf.out_block_size = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "cbs" => if let Some(m) = get_u(val) {
+                    cf.cnv_block_size = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "skip" => if let Some(m) = get_u(val) {
+                    cf.skip = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "seek" => if let Some(m) = get_u(val) {
+                    cf.seek = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "count" => if let Some(m) = get_u(val) {
+                    cf.count = m;
+                } else {
+                    err(format!("invalid {}: {}", key, val));
+                },
+                "conv" => for elem in val.split(",") {
+                    match Conf::get_flag(elem) {
+                        Conf::INVALID_FLAG => err("invalid flag passed to conv"),
+                        flag => cf.set_flag(flag),
+                    }
+                },
+                inv => err(format!("invalid argument: {}", inv)),
+            }
+        } else {
+            err(format!("invalid argument: {}", arg));
+        }
+    }
+
+    // Behaviour is unspecified. We choose to terminate.
```

Comment:
Does the GNU version terminate as well?

Reply:
I haven't read yet.

Reply:
Just an FYI but don't read the GNU source as it's licensed under the GPL.